web/src/app/util/SpeedDial.js

```diff
@@ -1,74 +1,67 @@
-import React from 'react'
+import React, { useState } from 'react'
 import p from 'prop-types'
-import SpeedDialAction from '@material-ui/lab/SpeedDialAction'
-import SpeedDialIcon from '@material-ui/lab/SpeedDialIcon'
-import SpeedDial from '@material-ui/lab/SpeedDial'
+import { SpeedDial, SpeedDialIcon, SpeedDialAction } from '@material-ui/lab'
+import { makeStyles } from '@material-ui/styles'
 
-export default class ScheduleNewOverrideFAB extends React.PureComponent {
-  static propsTypes = {
-    label: p.string.isRequired,
-    actions: p.arrayOf(
-      p.shape({
-        icon: p.element.isRequired,
-        onClick: p.func.isRequired,
-        label: p.string.isRequired,
-      }),
-    ).isRequired,
-  }
+const useStyles = makeStyles(theme => ({
+  speedDial: {
+    position: 'fixed',
+    bottom: '2em',
+    right: '2em',
+    zIndex: 9001,
+  },
+  staticTooltipLabel: {
+    whiteSpace: 'nowrap',
+  },
+}))
 
-  state = {
-    open: false,
-  }
+export default function CustomSpeedDial(props) {
+  const [open, setOpen] = useState(false)
+  const classes = useStyles()
 
-  shownState = false
-  _shownTimeout = -1
+  const doToggle = () => setOpen(!open)
+  const doOpen = () => setOpen(true)
+  const doClose = () => setOpen(false)
 
-  render() {
-    if (this.state.open !== this.shownState) {
-      clearTimeout(this._shownTimeout)
-      this._shownTimeout = setTimeout(() => {
-        this.shownState = this.state.open
-      }, 350)
-    }
-    const doToggle = () => this.setState({ open: !this.shownState })
-    const doOpen = () => this.setState({ open: true })
-    const doClose = () => this.setState({ open: false })
-    return (
-      <SpeedDial
-        ariaLabel={this.props.label}
-        ButtonProps={{
-          'data-cy': 'page-fab',
-        }}
-        icon={<SpeedDialIcon />}
-        onClick={doToggle}
-        onClose={doClose}
-        onMouseEnter={doOpen}
-        onMouseLeave={doClose}
-        open={this.state.open}
-        style={{
-          position: 'fixed',
-          bottom: '2em',
-          right: '2em',
-          zIndex: 9001,
-        }}
-      >
-        {this.props.actions
-          .slice()
-          .reverse()
-          .map((action, idx) => (
-            <SpeedDialAction
-              key={idx}
-              icon={action.icon}
-              tooltipOpen
-              tooltipTitle={action.label}
-              aria-label={action.label}
-              onClick={() => {
-                this.setState({ open: false })
-                action.onClick()
-              }}
-            />
-          ))}
-      </SpeedDial>
-    )
-  }
+  return (
+    <SpeedDial
+      ariaLabel={props.label}
+      FabProps={{
+        'data-cy': 'page-fab',
+      }}
+      icon={<SpeedDialIcon />}
+      onClick={doToggle}
+      onClose={doClose}
+      onMouseEnter={doOpen}
+      onMouseLeave={doClose}
+      open={open}
+      className={classes.speedDial}
+    >
+      {props.actions.map((action, idx) => (
```

Comment:
IIRC the speed dial menu displays things in reverse order, which is why we were switching them here. 

That way the prop-order matches display-order when using the `SpeedDial` component